pkg/pgmodel/end_to_end_tests/query_integration_test.go

```diff
@@ -2,26 +2,56 @@ package end_to_end_tests
 
```

Comment:
Hmm one more nit. Maybe these added tests are better in a separate file? This file is already long and checking promql queries vs remote_read stuff seems like a good delineation to split things out along.